avogadro/core/molecule.h

```diff
@@ -680,6 +710,31 @@ inline bool Molecule::setFormalCharge(Index atomId, signed char charge)
   return false;
 }
 
+inline Vector3ub Molecule::color(Index atomId) const
+{
+  return atomId < m_colors.size() ? m_colors[atomId] : Vector3ub(0, 0, 0);
+}
+
+inline bool Molecule::setColors(const Core::Array<Vector3ub>& colors)
+{
+  if (colors.size() == atomCount()) {
+    m_colors = colors;
+    return true;
+  }
+  return false;
+}
+
+inline bool Molecule::setColor(Index atomId, Vector3ub charge)
+{
+  if (atomId < atomCount()) {
+    if (atomId >= m_formalCharges.size())
+      m_colors.resize(atomCount(), {0, 0, 0});
```

Comment:
Here, it might be better to iterate through added colors, and default to the standard RGBs for the element.